annlite/core/index/hnsw/index.py

```diff
@@ -19,6 +22,7 @@ def __init__(
         ef_construction: int = 200,
         ef_search: int = 50,
         max_connection: int = 16,
+        using_pq: Optional['BaseTrainedPQ'] = None,
```

Comment:
I think the `HNSWIndex` interface should get only parameters, and the creation of Internal objects should be handled directly here. 

So here you should have the constructor parameters of the `BaseTrainedPQ` if desired?

Reply:
The backend needs a trained PQ. So the well-formatted parameters here are needed. But various PQ algorithms may need different parameters to construct and have distinct operations to run. That's why a class is passed here, it gives more freedom to users instead of only using the PQs we have supported.
(Again, it's over-designed here if we assume users don't want to customize their own PQ for the HNSW backend)